nixos/modules/system/etc/build-composefs-dump.py

```diff
@@ -61,7 +61,17 @@ def __init__(
         self.path = path
         self.size = size
         self.filetype = filetype
-        self.mode = mode
+
+        # We need to pad the mode, because we will later use the concatentation
+        # filetype|mode, which assumes that the mode has a length of 4.
+        match len(mode):
+            case 3:
+                self.mode = "0" + mode
+            case 4:
+                self.mode = mode
```

Comment:
I do wonder:
Wouldn't `self.mode = f"{mode:0>4}"` be more explicit of what is going on? We are padding, we should use the padding functions provided by python.